Update operation docs (#95)
kyma-project/kyma-environment-broker: docs/03-03-runtime-operations.md

None of the deprovisioning steps should block the entire deprovisioning operation. Use the `RetryOperationWithoutFail` function from the `DeprovisionOperationManager` struct to skip a step in case of a retry timeout. Set a 5-minute, at the most, timeout for retries in a step.
There is only one step, which fails the operation - `Check_Runtime_Removal`. It fails the operation in case of a timeout when checking for the Provisioner to remove the shoot. 
Once the step is successfully executed, it isn't retried (every deprovisioning step is defined in a separate stage). If a step has been skipped due to a retry timeout or error, a [Cron Job](03-16-deprovision-retrigger-cronjob.md) tries to deprovision all remaining Kyma runtime dependencies again at a scheduled time.
All deprovisioning steps can be found in the [deprovisioning.go file](../cmd/broker/deprovisioning.go).

**NOTE:** The timeout for processing this operation is set to `24h`.

## Update

The update process is triggered by [an OSB API update operation](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#updating-a-service-instance) request.

> All updating steps can be found in the [update.go file](../cmd/broker/update.go).

Reviewer: ```suggestion
You can find all the updating steps in the [update](../cmd/broker/update.go) file.
```

## Upgrade Cluster

Upgrade cluster process is triggered by upgrade cluster orchestration.

> All upgrading cluster steps can be found in the [upgrade_cluster.go file](../cmd/broker/upgrade_cluster.go).

Reviewer: ```suggestion
You can find all the upgrading cluster steps in the [upgrade_cluster](../cmd/broker/upgrade_cluster.go) file.
```

## Upgrade Kyma

Upgrade kyma process is triggered by upgrade kyma orchestration.

> All upgrading kyma steps can be found in the [upgrade_kyma.go file](../cmd/broker/upgrade_kyma.go).

Reviewer: ```suggestion
You can find all the upgrading kyma steps in the [upgrade_kyma](../cmd/broker/upgrade_kyma.go) file.
```

## Provisioning

> Provisioning process is executed when the instance is being created and when an unsuspension is triggered.

Reviewer 2: ```suggestion
Provisioning process is executed when the instance is being created or when an unsuspension is triggered.
```

Each provisioning step is responsible for a separate part of preparing Kyma runtime. For example, in a step you can provide tokens, credentials, or URLs to integrate SAP BTP, Kyma runtime with external systems.
All provisioning steps can be found in the [provisioning.go file](../cmd/broker/provisioning.go).

**NOTE:** The timeout for processing this operation is set to `24h`.

## Deprovisioning

Each deprovisioning step is responsible for a separate part of cleaning Kyma runtime dependencies. To properly deprovision all the dependencies, you need the data used during the Kyma runtime provisioning. The first step finds the previous operation and copies the data.

None of the deprovisioning steps should block the entire deprovisioning operation. Use the `RetryOperationWithoutFail` function from the `DeprovisionOperationManager` struct to skip a step in case of a retry timeout. Set a 5-minute, at the most, timeout for retries in a step.

> There is only one step, which fails the operation - `Check_Runtime_Removal`. It fails the operation in case of a timeout when checking for the Provisioner to remove the shoot. 

Author: There is one step which is an exception. All others should not fail the operation in case of an error. That is a general suggestion or a rule. It is not strict rule, but writing a code which fails the deprovisioning operation should be done carefully. If someone writes a piece of code which fails deprovisioning operation - must know the consequences.

> Once the step is successfully executed, it isn't retried (every deprovisioning step is defined in a separate stage). If a step has been skipped due to a retry timeout or error, a [Cron Job](03-16-deprovision-retrigger-cronjob.md) tries to deprovision all remaining Kyma runtime dependencies again at a scheduled time.

Author: Steps are grouped. Such group is called a stage. In the provisioning process, there are some groups which contains more than one step, but in the deprovisioning process there is always one step in a stage. The general rule is that if stage is finished (succeeded) then will never be retried.

> Once the step is successfully executed, it isn't retried (every deprovisioning step is defined in a separate stage). If a step has been skipped due to a retry timeout or error, a [Cron Job](03-16-deprovision-retrigger-cronjob.md) tries to deprovision all remaining Kyma runtime dependencies again at a scheduled time.

Author: 2. it is about any step
